### ISSUE TITLE
Deal properly with uppercase .GIT file basenames on web editor like for lowercase .git

### ISSUE THREAD

**Issue by cirosantilli** [isaacs/github#318]
Either:
-   on web UI blob create / rename, validate it nicely like for the lowercase `.git`.
  
  If you try to create a file named `.git` file from the web editor, it automatically removes that path component / deactivates the commit button.
  
  But if you use `.GIT` with upper case the creation fails less gracefully.
-   allow uppercase `.GIT`. I think this is the best possibility. 
  
  It is currently blocked on push and API.
  
  Git itself allows it, so it is an unnecessary restriction.
  
  GitHub can already currently deal with case sensitive files, e.g.: https://github.com/cirosantilli/test/blob/54203288c929edca978088d51b55b6b5ac0b3eca/CASE and https://github.com/cirosantilli/test/blob/54203288c929edca978088d51b55b6b5ac0b3eca/case
  
  GitLab allows it: https://gitlab.com/cirosantilli/test-GIT/tree/master

**Comment by stuartpb** [comment]
Yeah, `.GIT` is a bad citizen in the cross-platform Git community, so I'm all in favor of blacklisting it (especially since that's what the mainline client does now by default). They should unify their solutions, though (not allowing the commit from the editor in realtime).

Out of curiosity, if you bypass the `.git` prevention JS to post a file creation, does it give you the same error as `.GIT` currently does?
